pkg/daemon/daemon.go

```diff
@@ -519,14 +519,11 @@ func (dn *Daemon) nodeStateSyncHandler(generation int64) error {
 	}
 
 	// restart device plugin pod
-	if reqDrain || latestState.Spec.DpConfigVersion != dn.nodeState.Spec.DpConfigVersion {
```

Comment:
@e0ne With your fix, the device plugin pod will be restarted in every reconcile loop. It will cause many unnecessary pod restarts. 

The issue you mentioned only happens when the `numVFs` is changed from 0 to a non-zero value, as we don't need to drain the node when VFs are fresh created https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/ac5db516cd1255f5b7fd2ed91ecb98b60e8ba2be/pkg/plugins/generic/generic_plugin.go#L177

So a better solution should be to add this condition to trigger the device plugin pod restart, instead of always triggering a restart.

Reply:
Actually, this code will be executed only if node state is changed: https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/pkg/daemon/daemon.go#L438